balancer-js/src/modules/joins/joins.module.ts

```diff
@@ -179,17 +181,19 @@ export class Join {
       .div((1e18).toString());
     node.outputReference = inputAmount.toString();
     console.log(
-      node.type,
-      node.address,
-      node.action,
-      `Inputs: ${inputAmount.toString()}`,
-      `OutputRef: ${node.outputReference}`,
-      node.proportionOfParent.toString()
+      `${node.type} ${node.address} prop: ${node.proportionOfParent.toString()}
+      ${node.action} (
+        Inputs: ${inputAmount.toString()}
+        OutputRef: ${node.outputReference}
+      )`
     );
   }
 
   createAaveWrap(node: Node, sender: string, recipient: string): string {
-    const childNode = node.children[0]; // TODO: check if it's possible to have more than one child at this type of node
+    // TODO: check if it's possible to have more than one child at this type of node
+    // This might be the case for joining with non-leaf tokens
+    // Can aaveWraps be applied to tokens other than stables? Is it possible for stables to be non-leaf nodes? If not, then there will always be 1 child for this type of node.
+    const childNode = node.children[0];
```

Comment:
Moving the discussion here @johngrantuk - I have a few questions:
1. Can aaveWraps be applied to tokens other than stables?
2. Is it possible for stables to be non-leaf nodes?
If both answers are no, then there will always be a single child for this type of node.

Reply:
1. Aave can have many different yield bearing tokens and I believe it should be possible to wrap all of those.
2. I think a stable will always be a leaf node